docs/spec/plugins.md

```diff
@@ -0,0 +1,117 @@
+Plugins
+=======
+
+Specification
+-------------
+
+NSM provides a way to extend its functionality by creating plugins. Here is a small glossary:
+
+- **Plugin** — a gRPC server that implements one or more Plugin Features to extend NSM functionality
+- **Plugin Feature** — a gRPC service definition that specifies methods related to one small piece of NSM functionality (for example, connection feature is only responsible for updating and validating a connection)
```

Comment:
Not sure if 'feature' is the best name for this thing, but the idea is to have one plugin that implements few plugin services. This plugin should be registered only once and should specify all implemented plugin services during registration. What is the best suitable name for this?

This naming also appears in the code here:
https://github.com/networkservicemesh/networkservicemesh/blob/2ece99e3af397d5c54b77a214bb5526dff3d872e/controlplane/pkg/apis/plugins/registry.proto#L7

and here:
https://github.com/networkservicemesh/networkservicemesh/blob/2ece99e3af397d5c54b77a214bb5526dff3d872e/controlplane/pkg/apis/plugins/registry.proto#L13

Reply:
Renamed to capabilities